Tab section pattern

jmuzina opened the pull request:
## Done

Builds the tab section pattern.

Includes #5694 in the diff. Please merge #5694 first.

### TODO

- [x] Fix Percy visual regressions
- [x] Polish code
- [x] Docs page

## QA

- Open [full width](https://vanilla-framework-5712.demos.haus/docs/examples/patterns/tab-section/full-width)
- Open [50/50](https://vanilla-framework-5712.demos.haus/docs/examples/patterns/tab-section/50-50)
- Open [25/75](https://vanilla-framework-5712.demos.haus/docs/examples/patterns/tab-section/25-75)
- Open [combined examples](https://vanilla-framework-5712.demos.haus/docs/examples/patterns/tab-section/combined) and make sure everything looks OK. Note that combined examples are not interactive, so you won't be able to interact with the tabs.
- Open [docs](https://vanilla-framework-5712.demos.haus/docs/patterns/tab-section) and review the tab section docs. 
- Open [what's new](https://vanilla-framework-5712.demos.haus/docs/whats-new) and review the changelog

### Check if PR is ready for release

If this PR contains Vanilla SCSS or macro code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention
  - if existing APIs (CSS classes & macro APIs) are not changed it can be a bugfix release (x.x.**X**)
  - if existing APIs (CSS classes & macro APIs) are changed/added/removed it should be a minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) or macros should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

kim-isaac commented:
Thanks @jmuzina ! Looks great and glad to see the basic section included.

Regarding @eliman11 's comment, I think it’s fine to keep as it is in the Vanilla documentation, since the one in the Figma mainly focuses on content and design aspects. I’ll include that in the design documentation instead.

The Codepen example with scrolling also looks fine to me, as I don’t expect this case to occur often on real pages.

And I have one comments from the visual side:

- [ ] it looks like the shallow padding between the heading part and tabs is missing - could we add that?

<img width="1180" height="228" alt="image" src="https://github.com/user-attachments/assets/df1dc7db-6e92-4608-9475-3e8fb42b0bf8" />
<img width="1200" height="218" alt="image" src="https://github.com/user-attachments/assets/779bc361-01e9-44ee-94e9-84ab3534c644" />


Lastly, as discussed in the MM, the equal heights block should be added to all three layout options in a future update, separate from the current scope. If there’s anything you can share for follow-up, that would be much appreciated. For future reference, I’m sharing here the [comment](https://github.com/canonical/vanilla-framework/issues/5587#issuecomment-3647361147) you left on the equal heights ticket as well.

Thank you so much!!

jmuzina commented:
> Very minor but under the padding variants, could we list "default" as the first item in the bulleted list as it's the default value?

@eliman11 Good call, adjusted - thanks!

>  it looks like the shallow padding between the heading part and tabs is missing - could we add that?

@kim-isaac I've added shallow padding between the heading/description/cta and tabs. Can you have a look and let me know if it's as you expect?
